api/mesh/v1alpha1/http_retry_on.proto

```diff
@@ -0,0 +1,18 @@
+syntax = "proto3";
+
+package kuma.mesh.v1alpha1;
+
+option go_package = "github.com/kumahq/kuma/api/mesh/v1alpha1";
+
+enum HttpRetryOn {
+  all_5xx = 0;
+  gateway_error = 1;
+  reset = 2;
+  connect_failure = 3;
+  envoy_ratelimited = 4;
+  retriable_4xx = 5;
+  refused_stream = 6;
+  retriable_status_codes = 7;
+  retriable_headers = 8;
+  http3_post_connect_failure = 9;
+}
```

Comment:
I think I'd embed this into `Http` message in retry.proto. Just like grpc

Also, if we don't want to document every field with an explanation here, let's at least drop a link to envoy docs that explains what is `refused_stream` , `retriable_headers` etc.